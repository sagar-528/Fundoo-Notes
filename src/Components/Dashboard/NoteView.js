import React, { Component } from 'react'
import {ScrollView, View, Text} from 'react-native';
import * as Keychain from 'react-native-keychain';
import * as Animatable from 'react-native-animatable';
import NoteViewStyle from '../../Styles/NoteView'
import NoteCard from './NoteCard'
import SQLiteServices from '../../../Service/SQLiteServices'
import NoteDataControllerServices from '../../../Service/NoteDataControllerServices'

export default class NoteView extends Component {
    constructor(props) {
        super(props);
        this.state = {
            userNotes : [],
       }
    }

     componentDidMount = async () =>{
        const credential = await Keychain.getGenericPassword();
        const UserCredential = JSON.parse(credential.password);
       
      await  NoteDataControllerServices.retrieveDataFromFirebase(UserCredential.user.uid)
      await  SQLiteServices.selectNoteByDeletedFromSQliteStorage(UserCredential.user.uid, 0)
                .then(async result => {
                    var temp = [];
                    if(result.rows.length != 0) {
                        for (let i = 0; i < result.rows.length; ++i)
                        temp.push(result.rows.item(i));
                        await this.setState({
                            userNotes : temp.reverse()
                        })
                    }                
                })
            .catch(error => console.log(error))
            console.log(this.state.userNotes)
    }

    render() {

        return (
            <ScrollView style = {NoteViewStyle.container}>
            <Animatable.View 
            style = {NoteViewStyle.list_conatiner}
            animation = "fadeInUpBig">
            {this.state.userNotes.length > 0 ?
                this.state.userNotes.map(note => (
                        <React.Fragment key = {note.note_id}>
                            {
                                    <NoteCard listView = {this.props.listView} notes = {note} noteKey = {note.note_id} navigation = {this.props.navigation}/>
                            }
                        </React.Fragment>
                    )) 
                    :
                    null
                }
            </Animatable.View>
        </ScrollView>
        )
    }
}
