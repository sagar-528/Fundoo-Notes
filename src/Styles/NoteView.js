import { StyleSheet } from 'react-native'

const NoteViewStyle = StyleSheet.create({
    
    container : {
        flex: 1,
        backgroundColor: 'white'
    },

    list_conatiner : {
        flexDirection: 'row', 
        flexWrap: 'wrap',
    }
})

export default NoteViewStyle;