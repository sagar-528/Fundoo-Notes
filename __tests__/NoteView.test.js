import React from 'react';
import {shallow} from 'enzyme';
import NoteView from '../src/Components/Dashboard/NoteView'

describe('test Note View', () => {
    
    it('should match to snapshot', () => {
        const component = shallow(<NoteView />)
        expect(component).toMatchSnapshot();
    })

    // it('When component did mount method should get user details and Update State', async () => {
    //     const component = shallow(<NoteView />)
    //     const instance = component.instance();
    //     await instance.componentDidMount()
    // })

})

