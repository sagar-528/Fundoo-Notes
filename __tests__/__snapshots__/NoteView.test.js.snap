// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Note View should match to snapshot 1`] = `ShallowWrapper {}`;
